src/value/borrowed.rs

```diff
@@ -141,12 +141,46 @@ impl<'val> SymbolToken for BorrowedSymbolToken<'val> {
 /// A borrowed implementation of [`Builder`].
 impl<'val> Builder for BorrowedElement<'val> {
     type Element = BorrowedElement<'val>;
+    type SymbolToken = BorrowedSymbolToken<'val>;
     type Sequence = BorrowedSequence<'val>;
+    type Struct = BorrowedStruct<'val>;
 
     fn new_null(e_type: IonType) -> Self::Element {
         BorrowedValue::Null(e_type).into()
     }
 
+    fn new_bool(bool: bool) -> Self::Element {
+        BorrowedValue::Boolean(bool).into()
+    }
+
+    fn new_string(str: &'static str) -> Self::Element {
+        BorrowedValue::String(str).into()
+    }
```

Comment:
I am not sure about this--are you able to tie the life to `'val`? If not (e.g. this requires GAT), then I would eliminate it and only generically provide the `String` based constructor (similarly for `clob` and `blob`).